odxtools/exceptions.py

```diff
@@ -19,5 +20,61 @@ class OdxWarning(Warning):
     """Any warning that happens during interacting with diagnostic objects."""
 
 
-# Mark DecodeError warnings as errors by default
+# Mark DecodeError and EncodeErrpr warnings as errors by default
 warnings.simplefilter("error", DecodeError)
+warnings.simplefilter("error", EncodeError)
+
+#: Specify whether violations of the ODX specification or failed
+#: assumptions in the ODX library code should cause the routine in
+#: question to abort or whether it should continue anyway. Be aware that
+#: in non-strict mode, the behavior of odxtools is undefined and it
+#: might start eating children...
+strict_mode = True
+
+
+def odxraise(message: Optional[str] = None, error_type: Type[Exception] = OdxError) -> NoReturn:
+    """
+    Raise an exception but only if in strict mode.
+
+    Also, convince type checkers that the exception is always raised.
+    """
+    if TYPE_CHECKING or strict_mode:
+        if message is None:
+            raise error_type()
+        else:
+            raise error_type(message)
```

Comment:
Current approach will make all error appear as if they originated here, caller should be the one where the first line number show up, use `stacklevel` argument to reflect this
```suggestion
            warnings.warn(message, error_type, stacklevel=2)
```

Reply:
> use stacklevel argument to reflect this

thanks for the hint, I wasn't aware of this so far :)